packages/hint-create-element-svg/src/hint.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ * @fileoverview Inform users that they need to use createElementNS to create SVG elements instead of createElement
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+import { ScriptEvents, ScriptParse } from '@hint/parser-javascript';
+
+import { Linter } from 'eslint';
+import * as ESTree from 'estree';
+
+import meta from './meta';
+import svgElements from './svgElements';
+
+const debug: debug.IDebugger = d(__filename);
+
+export default class CreateElementSvgHint implements IHint {
+
+    public static readonly meta = meta;
+
+    public constructor(context: HintContext<ScriptEvents>) {
+        const linter = new Linter();
+
+        /*
+         * Check to see if any call expression in the AST has a callee with
+         * a 'createElement' property. If yes, check if that call expression's
+         * first argument is an SVG element. In that case, report an error as
+         * the correct identifier property for creating SVG elements is
+         * 'createElementNS'
+         */
+        linter.defineRule('svg-create', {
+            create(eslintContext) {
+                return {
+                    CallExpression(node) {
+                        const exp = node as ESTree.CallExpression;
+                        const args = exp.arguments;
+                        const callee = exp.callee as ESTree.MemberExpression;
+                        const property = callee !== undefined && callee !== null ? callee.property as ESTree.Identifier : null;
+
+                        if (property !== null && property.name === 'createElement' && args.length >= 1 && args[0].type === 'Literal') {
+                            const arg = args[0] as ESTree.Literal;
+
+                            if (arg.value !== undefined && arg.value !== null && svgElements.includes(arg.value.toString().toLowerCase())) {
+                                eslintContext.report({
+                                    messageId: 'avoidElement',
+                                    node
+                                });
+                            }
+                        }
+                    }
+                };
+            },
+            meta: { messages: { avoidElement: 'Avoid using createElement to create SVG elements; use createElementNS instead' } }
+        });
+
+        const validateScript = async (scriptData: ScriptParse) => {
+
+            debug(`Validating hint create-element-svg`);
+
+            const sourceCode = scriptData.sourceCode;
+            const results = linter.verify(sourceCode, { rules: { 'svg-create': 'error' } });
+
+            for (const result of results) {
+                await context.report(scriptData.resource, result.message);
```

Comment:
Looks like we're not passing along location information here. Not sure if there's a good place to get it off of `result` or not (maybe @molant knows)...
```suggestion
                await context.report(scriptData.resource, result.message, { location });
```

Reply:
I think that ESLint provides that information already. What I don't know is what happens if the script is inline with the HTML:

```html
<html>
<head>
  <script>
document.createElement('svg');
  </script>
</head>
</html>
```

Hopefully ESlint reports line 4 but we should cofirm this is the case (although maybe this just happens when using [`eslint-plugin-html`](https://www.npmjs.com/package/eslint-plugin-html).

Reply:
This was one thing I wasn't sure about ... result object has the location information, and I think report accepts location object, so I was thinking of supplying that here, but wasn't sure if eslint automagically does that stuff when it reports back?

Also, what's the best way to test this hint out against a real page (not test case) to see how it is reported back to a real user?

Reply:
Enable this in packages/hint/.hintrc and disable the rest of the rules. You can also install the vs code extension and it should detect all the rules and packages so you can create some js that uses it (you will still need a .hintrc file in the root)